olmo/model.py

```diff
@@ -1747,7 +1753,7 @@ def from_checkpoint(
         if checkpoint_type == CheckpointType.unsharded:
             # Initialize model (always on CPU to start with so we don't run out of GPU memory).
             model_config.init_device = "cpu"
-            model = OLMo(model_config)
+            model = OLMo(model_config).to(device)
```

Comment:
Did we talk about this? I'm pretty sure that doing this `.to(device)` thing will OOM for large distributed models.